File: 백준/Silver/1940. 주몽/주몽.py
import sys

n = int(sys.stdin.readline())
m = int(sys.stdin.readline())
arr = list(map(int, sys.stdin.readline().split()))

arr = sorted(arr)
cnt = 0
left, right = 0, len(arr)-1

while left < right:
    num = arr[left] + arr[right]

    if num < m:
        left += 1
    elif num > m:
        right -= 1
    else:
        cnt += 1
        left += 1
        right -= 1

sys.stdout.write(f"{cnt}\n")

File: 백준/Silver/1940. 주몽/주몽.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);

    int n, m;
    cin >> n >> m;
    int arr[n];

    for (int i = 0; i < n; i++)
        cin >> arr[i];

    sort(arr, arr+n);

    int cnt = 0;
    int left = 0, right = n-1;
    while (left < right) {
        int num = arr[left] + arr[right];

        if (num < m) left++;
        else if (num > m) right--;
        else {
            cnt++;
            left++;
            right--;
        }
    }

    cout << cnt << "\n";
    
    return 0;
}